src/btree/bt_read.c

```diff
@@ -530,9 +557,12 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
 	 * Don't free WT_REF.page_las, there may be concurrent readers.
 	 */
 	if (final_state == WT_REF_MEM && ref->page_las != NULL &&
-	    (!ref->page_las->skew_newest || ref->page_las->has_prepares))
+	    (!ref->page_las->skew_newest || ref->page_las->has_prepares)) {
 		WT_ERR(__wt_las_remove_block(
 		    session, ref->page_las->las_pageid));
+		__wt_free(session, ref->page_las->birthmarks);
```

Comment:
I don't get this part.
Isn't this array of birthmarks supposed to live as long as the `WT_REF` is alive? How come we're freeing it after instantiating.

Reply:
We are supposed to keep the birthmarks as long as there is any history for the page. In this case, here we have discarded the history and hence do not need it anymore. Later tickets might get rid of the code to read lookaside in the first place, but for now, we need to free both birthmarks and page_las when we discard the history.